rust/gui-client/src-tauri/src/client/network_changes/linux.rs

```diff
@@ -17,29 +19,57 @@ pub(crate) fn check_internet() -> Result<bool> {
     Ok(true)
 }
 
-pub(crate) struct Worker {}
+pub(crate) struct Worker {
+    interval: Interval,
+}
 
 impl Worker {
     pub(crate) fn new() -> Result<Self> {
-        Ok(Self {})
+        Ok(Self {
+            interval: create_interval(),
+        })
     }
 
     pub(crate) fn close(&mut self) -> Result<()> {
         Ok(())
     }
 
-    pub(crate) async fn notified(&self) {
-        futures::future::pending().await
+    pub(crate) async fn notified(&mut self) {
+        loop {
+            self.interval.tick().await;
+            tracing::trace!("Checking for network changes");
```

Comment:
I didn't implement this one, but this is where the Windows Client watches for loss / gain of Internet.

Reply:
is it worth adding a note/comment here?

Reply:
Yeah I'll add a note and just change it back to an infinite `future::pending`